### DISCUSSION TITLE
Wizard - Set initial state for all steps

### DISCUSSION THREAD

**Discussion by edhollinghurst** [data-driven-forms/react-forms#1335] [Need help] [answered]
Just wondered if it's possible for the initial state for all Wizard steps to be set on form render?

I was hoping this would be possible using `registeredFieldsHistory` but I haven't had any luck using this.

![SCR-20221101-pl2](https://user-images.githubusercontent.com/2844785/199309770-141af083-3c5c-450b-8b0c-18084b150f8a.png)

Code Sandbox example can be viewed here:
https://codesandbox.io/s/react-form-renderer-wizard-s3rdj1?file=/src/App.js

Many thanks!

**Comment by Hyperkid123** [comment, marked as answer]
@edhollinghurst Hi. Yes, it is very much possible. You have to use the `initialValues` prop on the form renderer to set initial values globally. The initial value used in a schema field will only appear in the form state once the field is registered to the form for the first time.

You can check this example, which is just a fork of yours: https://codesandbox.io/s/react-form-renderer-wizard-forked-o0hr15
The only change I made is taking the initialValues from your fields in the schema and passing them as a prop to the renderer.

![screenshot-codesandbox io-2022 11 02-09_00_10](https://user-images.githubusercontent.com/22619452/199432532-2927d3c5-f8a8-4256-a487-bf9e4693be8a.png)

**Reply by edhollinghurst** [reply]
Thanks so much for your quick reply and for all of your efforts with this project. 🙏
